kip-0037.md

```diff
@@ -0,0 +1,207 @@
+3. Error Handling:
+   - MUST return a clear error if account information cannot be retrieved due to configuration issues or unsupported networks.
+
+# Examples
+
+### Example request
+
+```json
+{
+  "id": 1,
+  "jsonrpc": "2.0",
+  "method": "kadena_getAccount_v1",
+  "params": {}
+}
+```
+
+### Example response
+
+```json
+{
+  "id": 1,
+  "jsonrpc": "2.0",
+  "result": {
+    "accountName": "k:35d381861d9a6cfb84bef47930b386d1a3b4f3742c8e87772a9e6b43c0e7b5f7",
+    "fungibleContract": "coin",
+    "chainAccounts": [
+      {
+        "chainId": "0",
+        "guard": {
+          "keys": [
+            "35d381861d9a6cfb84bef47930b386d1a3b4f3742c8e87772a9e6b43c0e7b5f7"
+          ],
+          "pred": "keys-all"
+        }
+      },
+      {
+        "chainId": "1",
+        "guard": {
+          "keys": [
+            "35d381861d9a6cfb84bef47930b386d1a3b4f3742c8e87772a9e6b43c0e7b5f7"
+          ],
+          "pred": "keys-all"
+        }
+      }
+    ]
+  }
+}
+```
+
+### Error Response
+
+```json
+{
+  "id": 1,
+  "jsonrpc": "2.0",
+  "error": {
+    "code": -32603,
+    "message": "Internal server error"
+  }
+}
+```
+
+### Response details
+
+Accounts should be returned per fungible contract, if the wallet knows about multiple fungibles held by one account then return the account for each one.
+
+If a wallet only works with single accounts, this method should return a single account.
+
+### Request by public key
+
+kip-0017 defines kadena_getAccounts_v1, which uses a public key received from walletconnect to retrieve accounts.
+
+## Example request by public key
+
+```json
+{
+  "id": 1,
+  "jsonrpc": "2.0",
+  "method": "kadena_getAccount_v1",
+  "params": {
+    "accounts": {
+      "publicKey": "60870392901728fe6bb773534d71272d0851b587cb91cb859a303b20f1025069",
+      "contract": "coin"
+    }
+  }
+}
+```
+
+## Example response by public key
+
+```json
+{
+  "id": 1,
+  "jsonrpc": "2.0",
+  "result": {
+    "accountName": "w:Gkx0oHalSq0TpNQjx7Q4UTgZteaQ1QwnBH3hR60VCuY:keys-all",
+    "fungibleContract": "coin",
+    "chainAccounts": [
+      {
+        "chainId": "0",
+        "guard": {
+          "keys": [
+            "60870392901728fe6bb773534d71272d0851b587cb91cb859a303b20f1025069",
+            "49a7943c8a29962d81142067da4f4fa98940fee89efc4780b86adef9546b493a"
+          ],
+          "pred": "keys-all"
+        }
+      }
+    ]
+  }
+}
+```
+
+## Backwards Compatibility
+
+- Legacy Methods: Wallets MAY support older methods for retrieving account information alongside kadena_getAccounts_v1. Legacy methods MUST be documented as deprecated and phased out over time.
+- Configuration Consistency: Implementations MUST ensure that the returned account information matches the wallet’s active configuration.
+
+## Security Considerations
+
+- Data Accuracy: Ensure that the returned account information is accurate and verified.
+- Event Notification: Emit account change events to keep dApps updated and prevent configuration mismatches.
```

Comment:
Modify this and reference the optional KIP that describes events. In that KIP you can add that WalletConnect supports events, and the wallet-adapter will implement those events on a best-effort basis. E.g. when a wallet provider has support for event-based communication this can then propagate to the dapp, or can be emulated by the wallet-adapter itself